src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -189,13 +190,21 @@ public sealed class ReadyToRunCodegenCompilation : Compilation
         /// </summary>
         private readonly JitConfigProvider _jitConfigProvider;
 
+        /// <summary>
+        /// We only need one CorInfoImpl per thread, and we don't want to unnecessarily construct them
+        /// because their construction takes a significant amount of time.
+        /// </summary>
+        private readonly ConditionalWeakTable<Thread, CorInfoImpl> corInfoImpls;
```

Comment:
What's the difference between having corInfoImpls as a field vs having it as a local like it was before?

Reply:
Having it as a local meant that for the next batch of methods we would clean up all the CorInfoImpls after the CWT went out of scope, which is unnecessary.

Reply:
Thanks